editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-execution-scope.tsx

```diff
@@ -106,16 +106,15 @@ export function createExecutionScope(
   }
   // TODO All of this is run on every interaction o_O
 
-  let topLevelJsxComponents: Map<string, UtopiaJSXComponent> = new Map()
+  let topLevelJsxComponents: Map<string | null, UtopiaJSXComponent> = new Map()
 
   // Make sure there is something in scope for all of the top level components
   fastForEach(topLevelElements, (topLevelElement) => {
     if (isUtopiaJSXComponent(topLevelElement)) {
       topLevelJsxComponents.set(topLevelElement.name, topLevelElement)
-      if (!(topLevelElement.name in topLevelComponentRendererComponentsForFile)) {
-        topLevelComponentRendererComponentsForFile[
-          topLevelElement.name
-        ] = createComponentRendererComponent({
+      const elementName = topLevelElement.name ?? 'default'
+      if (!(elementName in topLevelComponentRendererComponentsForFile)) {
+        topLevelComponentRendererComponentsForFile[elementName] = createComponentRendererComponent({
```

Comment:
can the elementName `'default'` clash with a user element name? I _think_ not because components must start with a capital letter

Reply:
I don't _believe_ so no.